salt/states/module.py

```diff
@@ -319,13 +319,10 @@ def _call_function(name, returner=None, **kwargs):
     arg_type, na_type, kw_type = [], {}, False
     for funcset in reversed(kwargs.get('func_args') or []):
         if not isinstance(funcset, dict):
-            kw_type = True
-        if kw_type:
-            if isinstance(funcset, dict):
-                arg_type += funcset.values()
-                na_type.update(funcset)
-            else:
-                arg_type.append(funcset)
+            arg_type.append(funcset)
+        elif isinstance(funcset, dict):
```

Comment:
It looks `else` is enough here.

Reply:
Yeah. Again, when you factor out the removed code, the preceding `if` is already doing `not isinstance(funcset, dict)`. This means that an `else` implicitly assumes that `funcset` _is_ a dict, making a second type check redundant.